app/src/shared/web3/web3Provider.js

```diff
@@ -21,9 +21,11 @@ type StreamrWeb3Options = {
 
 export class StreamrWeb3 extends Web3 {
     isLegacy: boolean
+    metamaskProvider: any
 
     constructor(provider: any, options: StreamrWeb3Options = {}) {
         super(provider)
+        this.metamaskProvider = provider
```

Comment:
Store the Metamask provider, `window.ethereum` or sometimes `web3.currentProvider` for older versions.

We should still be fine using the `currentProvider` that comes from the `StreamrWeb3` wrapped object.

Reply:
Not sure if it's worth bothering with legacy support, it would be for a very old metamask version.

Reply:
Probably not but does no harm and should work so I think it's good to leave it in.